### PR TITLE
update eslint configuration

### PR DESCRIPTION
**Pull request by benschac**
## What does this PR do and why?

This is the last PR in the diff stack and the spiciest.

**What** 
ESlint had a major breaking change under the hood in RCA (https://github.com/facebook/create-react-app/blob/main/CHANGELOG.md#highlights) from v4 to v5. CRA now uses Eslint 8. Additionally, there are breaking changes in the react-supported ESLint plugins. It's a pretty common problem when upgrading: https://stackoverflow.com/questions/58065765/eslint-jest-globals-environment-key-unknown

- Deleted the old ESlint modules and ran through the ESlint getting started flow. https://eslint.org/docs/user-guide/getting-started.
- Saved the old config and added back as many rules as possible without breaking the project.

**Why**
ESLint needed to be upgraded because of the breaking change to RCA in `react-scripts` v5 using ESLint version 8.

**Additional Notes**
Why this change is spicy: I upgraded ESLint and there are a bunch of linting rules that the old configuration didn't have by default, which now ships with the default configuration:

```
"eslint:recommended",
"plugin:react/recommended",
"plugin:@typescript-eslint/recommended",
 ```
 
 The goal of this PR was to just get the project running without errors. I didn't make any assumptions on rules that the team would or wouldn't want to include in the project because I know have enough context into what y'all would like to have linted and then fixed, so I just added eslint comments to turn off errors I wasn't sure the team wanted to support.

For the team: cc @johnnyd-eth @peripheralist @aeolianeth -- which ESLint ignore comments should we fix and support moving forward?

I can add those fixes in another PR or just fix them here.

## Screenshots or screen recordings

N/A, but the project now runs like it normally would without any errors.

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

**Comment by benschac** [comment]
Current dependencies on/for this PR:
* main
  * **PR #867** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/867" target="_blank">Graphite</a> 
    * **PR #868** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/868" target="_blank">Graphite</a> 
      * **PR #869** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/869" target="_blank">Graphite</a> 
        * **PR #870** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/870" target="_blank">Graphite</a> 
          * **PR #871** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/871" target="_blank">Graphite</a> 
            * **PR #876** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/876" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/876?utm_source=stack-comment).

File: package.json
Comment (benschac) [inline package.json:101]: I'm not 100% if we need to install ESLint here because react-scripts is using ESLint8 under the hood.

Comment (benschac) [reply package.json:101]: I'm going to test this locally, and see if it can be removed.

File: .eslintrc.json
Comment (benschac) [inline .eslintrc.json:25]: These come by default with `plugin:react/recommended` I just turned them off, because they caused some errors.

File: .eslintrc.json
Comment (benschac) [inline .eslintrc.json:31]: These come by default with plugin:react/recommended I just turned them off, because they caused some errors.

**Review by johnnyd-eth — APPROVED** [review]
Very nice @benschac thanks for picking this up. All LGTM but I'm gonna defer to @aeolianeth  on the question of ignore comments - think he'll have a better answer than myself. Will also wait to get another set of eyes before we merge

File: package.json
Comment (benschac) [reply package.json]: removed

**Comment by aeolianeth** [comment]
@benschac thanks for the thoughtfulness with this one!

I'm fine with disabling those new rules at the config level.

I think we should disabling eslint for specific lines though. Most of these look pretty trivial to fix (although I could be wrong) - would it be much of a lift to fix these and remove the `disable-next-line` comments?

**Comment by benschac** [comment]
Yeah, I can add another PR to the stack with the fixes.

File: src/components/shared/IssueTicketsButton.tsx
Comment (benschac) [inline src/components/shared/IssueTicketsButton.tsx:1]: This component made its way in here during a re-base. I need to fix this. Sorry!

**Comment by benschac** [comment]
This needs a bit more work. I wish there were an easy way to demote back into a WIP PR.

**Pull request merged**
